util/dvsim/Modes.py

```diff
@@ -531,21 +549,31 @@ def create_regressions(regdicts, sim_cfg, tests):
                 if sim_mode_obj.name in sim_cfg.en_build_modes:
                     continue
 
-                # Merge the build and run opts from the sim modes
+                # Merge the build and run cmds & opts from the sim modes
+                regression_obj.pre_build_cmds.extend(
+                    sim_mode_obj.pre_build_cmds)
+                regression_obj.post_build_cmds.extend(
+                    sim_mode_obj.post_build_cmds)
                 regression_obj.build_opts.extend(sim_mode_obj.build_opts)
+                regression_obj.pre_run_cmds.extend(sim_mode_obj.pre_run_cmds)
+                regression_obj.post_run_cmds.extend(sim_mode_obj.post_run_cmds)
                 regression_obj.run_opts.extend(sim_mode_obj.run_opts)
```

Comment:
There's an awful lot of repetition here (and in the other code in Modes.py). I wonder whether we could tidy it up. Would you be willing to review a follow-up commit from me that did so?

Reply:
I am scared to say yes, after looking at the monstrosity in #3779 :-D

I'd prefer to document it and take the "don't fix if it ain't broke" approach. We can address it after tape-out. Right now, there are more pressing matters at hand.

Reply:
I'll leave dvsim changes to you.